### PR TITLE
Created the fun-explorer crate

### PR DESCRIPTION
**Pull request by junkil-park**
This PR is submitted to share some files and tools with people, and not necessarily to merge into the master.

This crate defines SimpleEnv to store and explore the information of all functions in the Libra Framework.
- The function information is loaded from the pre-created JSON file (e.g., `functions.json`).
- SimpleEnv is `String`-based.  `SimpleEnv` uses the `String` type as the basic type all over the places. However, it should cause no serious performance issue because the size of the Libra Framework is not too large.

### CODE REVIEW

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M serde_json 1.0.59 (direct third-party, crates.io)
    * version upgraded from 1.0.58
    * (unchanged features: default, std)
```

File: language/move-prover/fun-explorer/functions.json [hunk 1]
```diff
@@ -0,0 +1,5346 @@
+[
```
Comment (mimoo) [inline language/move-prover/fun-explorer/functions.json:1]: does this need to be checked in the repo? (If this is automatically generated)

Comment (mimoo) [reply language/move-prover/fun-explorer/functions.json:1]: (perhaps you can generate that with a `build.rs`?)

File: Cargo.lock [hunk 2]
```diff
@@ -5926,9 +5934,9 @@ dependencies = [
 
 [[package]]
 name = "serde_json"
-version = "1.0.58"
+version = "1.0.59"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "a230ea9107ca2220eea9d46de97eddcb04cd00e92d13dda78e478dd33fa82bd4"
+checksum = "dcac07dbffa1c65e7f816ab9eba78eb142c6d44410f4eeba1e26e4f5dfa56b95"
```
Comment (mimoo) [inline Cargo.lock:5939]: can you leave this at version 58 so that you don't need a TCB dep approval?

**Comment by davidiw** [comment]
Closing out old PR.

**Pull request closed without merging**